WooCommerce/Classes/Authentication/Store Creation/Progress/StoreCreationProgressViewModel.swift

```diff
@@ -0,0 +1,165 @@
+import UIKit
+
+final class StoreCreationProgressViewModel: ObservableObject {
+    enum Progress: Float, CaseIterable {
+        case creatingStore = 0.0
+        case buildingFoundations = 1.0
+        case organizingStockRoom = 2.0
+        case applyingFinishingTouches = 3.0
+        case finished = 4.0
+    }
+
+    @Published private(set) var progress: Progress = .creatingStore {
+        didSet {
+            progressValue = progress.rawValue
+        }
+    }
+
+    let totalProgressAmount = StoreCreationProgressViewModel.Progress.finished.rawValue
+
+    @Published private(set) var progressValue: Float = StoreCreationProgressViewModel.Progress.creatingStore.rawValue
+
+    var title: String {
+        progress.title
+    }
+
+    var subtitle: AttributedString {
+        var prefix = AttributedString(Localization.funWooFact)
+        prefix.font = .headline
+        prefix.foregroundColor = .init(.text)
+
+        var attributedText = AttributedString(progress.subtitle)
+        attributedText.font = .body
+        attributedText.foregroundColor = .init(.text)
+
+        return prefix + " " + attributedText
+    }
+
+    private var animationTimer: Timer?
+    private let progressViewAnimationTimerInterval: TimeInterval
+    private let incrementProgressValueBy: Float
+
+    /// - Parameters:
+    ///   - incrementInterval: Interval at which progress will be incremented to next case
+    ///   - progressViewAnimationTimerInterval: Animation timer interval DI for unit test purposes.
+    init(incrementInterval: TimeInterval,
```

Comment:
nit: I spent some time understanding this parameter, since the actual trigger to the next case is still the  `incrementProgress` function.

How about renaming this so that it's an "approximate" time interval for the current progress state?

Reply:
Renamed in 6c32661e302bdbe9ff0bd79a369fd724b7ceb48a. Let me know if you think we can use a different name.